frontend/src/lib/components/Annotations/Annotations.tsx

```diff
@@ -24,38 +25,42 @@ export function Annotations({
     leftExtent,
     interval,
     topExtent,
+    insightShortId,
     insightId,
     onClick,
     color,
     accessoryColor,
     onClose,
     graphColor,
     currentDateMarker,
-}: AnnotationsProps): JSX.Element[] {
-    const { diffType, groupedAnnotations } = useValues(annotationsLogic({ insightId }))
+}: AnnotationsProps): JSX.Element {
+    const logic = annotationsLogic({ insightId })
+    const { diffType, groupedAnnotations } = useValues(logic)
+    const { createAnnotation, deleteAnnotation, deleteGlobalAnnotation, createGlobalAnnotation } = useActions(logic)
 
-    const { createAnnotation, createAnnotationNow, deleteAnnotation, deleteGlobalAnnotation, createGlobalAnnotation } =
-        useActions(annotationsLogic({ insightId }))
+    const onCreate =
+        (date: string) =>
+        (input: string, applyAll: boolean): void => {
+            if (applyAll) {
+                createGlobalAnnotation(input, date, insightShortId)
```

Comment:
I think replacing `insightShortId` here with just `insightId` will simplify things down the line and avoid one `getInsightId` call?